base/log_keys.go

```diff
@@ -0,0 +1,105 @@
+package base
+
+import (
+	"sync/atomic"
+)
+
+// LogKey is a bitfield of log keys.
+type LogKey struct {
+	flag uint32
+}
+
+// Values for log keys.
+const (
+	// KEY_NONE is shorthand for no log keys.
+	KEY_NONE uint32 = 0
+
+	// KEY_ALL is a wildcard for all log keys.
+	KEY_ALL uint32 = 1 << iota
```

Comment:
Based on our conversation, I think I understand the motivation for KEY_ALL, and I assume KEY_NONE is just to make LogKey==0 a bit more self-descriptive?

Reply:
Yeah, just to re-iterate on here: `KEY_ALL` has its own bit so that it can be toggled on/off without wiping out existing log keys (this behaviour is tested at the bottom of `TestLogKey`)

`KEY_NONE` is purely for consistency.